### PR TITLE
2021-05-06 v. 1.0.7: added in-app update feature

### LINKED ISSUE #41: Add In-app update feature
https://developer.android.com/guide/playcore/in-app-updates

### PR DESCRIPTION
**Pull request by vipozdn**
1.Added:
- play core library dependency in app/build.gradle
- flexible in-app update feature in MainActivity

## Checklist

__Common__

- [x] I am ran the app before creating PR
- [x] I am ran all tests before creating PR

__UI__

- [x] I am ran the app for visual checks.

__Logic__

- [x] I am tested basic app functionality.

## Changes

Describe all changes here:

- First;
- Second;
- Third;
- ...

## Comments

Describe all additional information here.

### CODE REVIEW

File: app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java [hunk 1]
```diff
@@ -27,6 +38,9 @@
     public static final String URI_FROM_EXTERNAL_APP
             = "com.smlnskgmail.jaman.hashcheckerlite.URI_FROM_EXTERNAL_APP";
 
+    private static final int MY_REQUEST_CODE = 1;
```
Comment (fartem) [inline app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java]: Please rename `MY_REQUEST_CODE` to `REQUEST_APP_UPDATE`.

File: app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java [hunk 2]
```diff
@@ -163,4 +194,49 @@ public void onBackPressed() {
...
+            e.printStackTrace();
+        }
+    }
+
+    // Displays the snackbar notification and call to action.
+    private void popupSnackbarForCompleteUpdate() {
+        Snackbar snackbar =
+                Snackbar.make(
+                        findViewById(android.R.id.content).getRootView(),
+                        "An update has just been downloaded.",
```
Comment (fartem) [inline app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java]: You need to move all strings in UI to `strings.xml` file in resources. I'm translating this sentences to other languages.

**Review by fartem — CHANGES_REQUESTED** [review]

**Review by fartem — APPROVED** [review]

**Review by fartem — APPROVED** [review]

**Comment by fartem** [comment]
@vipozdn I'm added you to Contributors list in README.md.

**Comment by vipozdn** [comment]
> 
> 
> @vipozdn I'm added you to Contributors list in README.md.

That's cool! Tnx!

**Pull request merged**